sympy/functions/elementary/piecewise.py

```diff
@@ -1032,6 +1032,9 @@ def _eval_rewrite_as_ITE(self, *args, **kwargs):
     def _eval_rewrite_as_KroneckerDelta(self, *args):
         from sympy import Ne, Eq, KroneckerDelta
 
+        class UnrecognizedCondition(Exception):
+            pass
```

Comment:
It would be nice to have the offending condition printed out.

Reply:
It shouldn't be possible for this exception to leak out of rewrite.

Reply:
It's still useful for debugging.